rust/cardano-blockchain-types/src/hashes.rs

```diff
@@ -0,0 +1,144 @@
+//! Cardano hashing functions
+
+use std::{fmt, str::FromStr};
+
+use anyhow::bail;
+use blake2b_simd::Params;
+use pallas_crypto::hash::Hash;
+
+/// Number of bytes in a blake2b 224 hash.
+pub const BLAKE_2B224_SIZE: usize = 224 / 8;
+/// `Blake2B` 224bit Hash
+pub type Blake2b224Hash = Blake2bHash<BLAKE_2B224_SIZE>;
+
+/// Number of bytes in a blake2b 256 hash.
+pub const BLAKE_2B256_SIZE: usize = 256 / 8;
+/// `Blake2B` 256bit Hash
+pub type Blake2b256Hash = Blake2bHash<BLAKE_2B256_SIZE>;
+
+/// Number of bytes in a blake2b 128 hash.
+pub const BLAKE_2B128_SIZE: usize = 128 / 8;
+/// `Blake2B` 128bit Hash
+pub type Blake2b128Hash = Blake2bHash<BLAKE_2B128_SIZE>;
+
+/// data that is a blake2b [`struct@Hash`] of `BYTES` long.
+///
+/// Possible values with Cardano are 32 bytes long (block hash or transaction
+/// hash). Or 28 bytes long (as used in addresses)
+#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct Blake2bHash<const BYTES: usize>(Hash<BYTES>);
+
+impl<const BYTES: usize> Blake2bHash<BYTES> {
+    /// Create a new `Blake2bHash` from a slice of bytes by hashing them.
+    #[must_use]
+    pub fn new(input_bytes: &[u8]) -> Self {
```

Comment:
Well, in my thinking, we are creating a `new` hash, and we do that from a set of bytes that need hashing.
Whereas `TryFrom` is converting a raw hash thats already in a byte array form.
Note, TryFrom will fail if the byte array isn't the correct size, and we are highly unlikely to be hashing byte buffers that are the exact size as the hash itself (it could happen, but its unlikely).

I think the comment suffices, but it maybe could be better.
If `TryFrom` was the same as `new` i wouldn't have `new` at all.  Because one should be using `try_from` or `from` to convert equivalent types, and `new` to create a new instance from the supplied data.

For me I think its an anti pattern to use `new` to convert types.